### PR TITLE
234 RQ support for datahandler 'query' operations

### PR DESCRIPTION
**Pull request by ags-tolson**
See #234 for the new file layout, mostly that `queue/queue.py` is the router that calls torque.submit or rq.submit depending on a new gips setting "TASK_QUEUE"; defaults to 'torque'.  See `scheduler.schedule_query` for the kind of rework that scheduler needs to support RQ (race condition, really not resolvable except by rework).  Commit messages below have more info:

### CODE REVIEW

File: gips/datahandler/logger.py [hunk 1]
```diff
@@ -33,8 +33,10 @@ def __new__ (cls):
     def __init__(self):
         self.extra = {}
         try:
+            # TODO this can't work for RQ
```
Comment (justinfisk) [inline gips/datahandler/logger.py]: should be replaced with queue.get_job_name()

File: gips/datahandler/logger.py [hunk 2]
```diff
@@ -33,8 +33,10 @@ def __new__ (cls):
...
             self.extra['jobid'] = os.environ['PBS_JOBID']
         except:
+            # TODO is this correct?
```
Comment (justinfisk) [inline gips/datahandler/logger.py]: I thought this was reasonable. Some things that log are not queued jobs -- so just use the script name (scheduler, geokitd). Idea being that the log needs to report something about what is doing the logging. 

But the try-except will probably not work without modification. It currently depends on failing on the lack of env var. Should be rolled into get_job_name and made clever.

File: gips/datahandler/queue/queue.py [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+
+def get_queue_module():
+    """Return the appropriate module based on GIPS setting TASK_QUEUE."""
+    if tq_setting() == 'torque':
+        from . import torque
+        return torque
+    if tq_setting() == 'rq':
+        from . import rq
+        return rq
+
```
Comment (justinfisk) [inline gips/datahandler/queue/queue.py]: What if neither?

File: gips/datahandler/worker.py [hunk 4]
```diff
@@ -226,3 +226,10 @@ def export_and_aggregate(job_id, start_ext, end_ext,
 
     if cleanup:
         shutil.rmtree(outdir)
+
+
+def look_busy(duration=5):
```
Comment (justinfisk) [inline gips/datahandler/worker.py]: I understand the usefulness of this for your testing, but should this be merged?

File: gips/datahandler/queue/queue.py (hunk 3 above)
Comment (ags-tolson) [inline gips/datahandler/queue/queue.py]: `tq_setting` is supposed to ensure that's impossible, but if you like I can `raise RuntimeError("Unreachable line reached")` or similar

File: gips/datahandler/logger.py (hunk 2 above)
Comment (ags-tolson) [inline gips/datahandler/logger.py]: Sorry that's a stale TODO, I later tested it and discovered it worked.  It's easy to take this out.

File: gips/datahandler/worker.py (hunk 4 above)
Comment (ags-tolson) [inline gips/datahandler/worker.py]: It looks like this PR will need a commit or two before merging so I'll take it out while I'm in there.

File: gips/datahandler/logger.py (hunk 1 above)
Comment (ags-tolson) [inline gips/datahandler/logger.py]: sounds good

**Pull request merged**